File: ios/snacks/src/OObject.swift
//
//  OObject.swift
//  snacks
//
//  Created by 谢月甲 on 16/2/21.
//  Copyright © 2016年 noear. All rights reserved.
//

import Foundation

public class OObject {
    public final var members:[String:ONode] = [String:ONode]();
    
    public func set(key:String,_ value:ONode){
        members[key] = value;
    }
    
    public func get(key:String)->ONode{
        return members[key]!;
    }
    
    public func contains(key:String)->Bool{
        return members.contains({ (k,v) -> Bool in
            return key == k;
        });
    }
    
    public func count()->Int{
        return members.count;
    }
    
    public func clear(){
        return members.removeAll();
    }
}

File: java/Snacks/src/main/java/noear/snacks/OObject.java
package noear.snacks;

import noear.snacks.ONode;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * Created by noear on 14-6-11.
 */
public class OObject {

    public final Map<String, ONode> members;

    public OObject(){
        members = new LinkedHashMap<String, ONode>();
    }

    public void set(String key, ONode value)
    {
        members.put(key,value);
    }

    public ONode get(String key)
    {
        return members.get(key);
    }

    public void rename(String key, String newKey) {
        ONode val = members.get(key);
        members.remove(key);
        members.put(newKey, val);
    }

    public void remove(String key){
        members.remove(key);
    }

    public boolean contains(String key)
    {
        return members.containsKey(key);
    }

    public int count() {
        return members.size();
    }

    public void clear(){
        members.clear();
    }
}


